diff --git a/torch/_decomp/decompositions.py b/torch/_decomp/decompositions.py
index 26043e78ce9..ed27d6cd73a 100644
--- a/torch/_decomp/decompositions.py
+++ b/torch/_decomp/decompositions.py
@@ -3331,6 +3331,17 @@ def upsample_bicubic2d_vec(
     scale_h, scale_w = scale_factors if scale_factors else (None, None)
     return upsample_bicubic2d_default(a, output_size, align_corners, scale_h, scale_w)
 
+@register_decomposition(aten._chunk_grad_outputs_efficient_attention)
+def  _chunk_grad_outputs_efficient_attention(query, key, value, is_causal=False):
+    M = query.size(2)
+    N = key.size(2)
+    grad_kv_needs_init = is_causal and N > M
+    is_aliased = torch._C._is_alias_of(query, key) and torch._C._is_alias_of(query, value)
+    equal_seq_len = query.size(2) == key.size(2)
+    q_v_same_head_dim = query.size(3) == value.size(3)
+    chunk_grad_outputs = not grad_kv_needs_init and equal_seq_len and q_v_same_head_dim and is_aliased
+    return chunk_grad_outputs
+
 
 @register_decomposition(aten.aminmax)
 @out_wrapper("min", "max")
