diff --git a/test/inductor/test_smoke.py b/test/inductor/test_smoke.py
index 2a247fddbe7..c226d61225e 100644
--- a/test/inductor/test_smoke.py
+++ b/test/inductor/test_smoke.py
@@ -5,12 +5,9 @@ import unittest
 import torch
 import torch._logging
 from torch._inductor.test_case import TestCase
-from torch.testing._internal.common_utils import IS_LINUX
-from torch.testing._internal.inductor_utils import (
-    GPU_TYPE,
-    HAS_CUDA_AND_TRITON,
-    HAS_GPU,
-)
+from torch.testing._internal.common_device_type import instantiate_device_type_tests
+from torch.testing._internal.common_utils import HardwareClassification
+from torch.utils._triton import has_triton
 
 
 class MLP(torch.nn.Module):
@@ -30,21 +27,31 @@ def _test_f(x):
 
 
 class SmokeTest(TestCase):
-    @unittest.skipIf(not HAS_GPU, "Triton is not available")
-    def test_mlp(self):
+    hw_classification = HardwareClassification.GENERIC
+
+    def test_compile_invalid_options(self):
+        with self.assertRaises(RuntimeError):
+            torch.compile(_test_f, mode="ha")
+
+
+class SmokeTestDevice(TestCase):
+    hw_classification = HardwareClassification.ACCELERATOR
+
+    @unittest.skipIf(not has_triton(), "requires triton")
+    def test_mlp(self, device):
         torch._logging.set_logs(
             dynamo=logging.DEBUG, inductor=logging.DEBUG, aot=logging.DEBUG
         )
 
-        mlp = torch.compile(MLP().to(GPU_TYPE))
+        mlp = torch.compile(MLP().to(device))
         for _ in range(3):
-            mlp(torch.randn(1, device=GPU_TYPE))
+            mlp(torch.randn(1, device=device))
 
         # set back to defaults
         torch._logging.set_logs()
 
-    @unittest.skipIf(not HAS_GPU, "Triton is not available")
-    def test_compile_decorator(self):
+    @unittest.skipIf(not has_triton(), "requires triton")
+    def test_compile_decorator(self, device):
         @torch.compile
         def foo(x):
             return torch.sin(x) + x.min()
@@ -54,17 +61,16 @@ class SmokeTest(TestCase):
             return x * x
 
         for _ in range(3):
-            foo(torch.full((3, 4), 0.7, device=GPU_TYPE))
-            bar(torch.rand((2, 2), device=GPU_TYPE))
+            foo(torch.full((3, 4), 0.7, device=device))
+            bar(torch.rand((2, 2), device=device))
 
-    def test_compile_invalid_options(self):
-        with self.assertRaises(RuntimeError):
-            torch.compile(_test_f, mode="ha")
+
+instantiate_device_type_tests(
+    SmokeTestDevice, globals(), except_for="cpu", allow_xpu=True
+)
 
 
 if __name__ == "__main__":
     from torch._inductor.test_case import run_tests
 
-    if IS_LINUX and HAS_GPU:
-        if (not HAS_CUDA_AND_TRITON) or torch.cuda.get_device_properties(0).major <= 5:
-            run_tests()
+    run_tests()
