diff --git a/test/cpython/v3_13/test_unary.py b/test/cpython/v3_13/test_unary.py
index a45fbf6bd6bc54b150c80902f09c8dcf0c0fc673..e2f8ba824a1c1a7aab9d14913d0412f6867544fa 100644
--- a/test/cpython/v3_13/test_unary.py
+++ b/test/cpython/v3_13/test_unary.py
@@ -1,8 +1,26 @@
+# ======= BEGIN Dynamo patch =======
+# Owner(s): ["module: dynamo"]
+
+# ruff: noqa
+# flake8: noqa
+
+# Test copied from
+# https://raw.githubusercontent.com/python/cpython/refs/tags/v3.13.5/Lib/test/test_unary.py
+
+import sys
+import torch
+import torch._dynamo.test_case
+import unittest
+from torch._dynamo.test_case import CPythonTestCase
+from torch.testing._internal.common_utils import run_tests
+
+# ======= END DYNAMO PATCH =======
+
 """Test compiler changes for unary ops (+, -, ~) introduced in Python 2.2"""
 
 import unittest
 
-class UnaryOpTestCase(unittest.TestCase):
+class UnaryOpTestCase(CPythonTestCase):
 
     def test_negative(self):
         self.assertTrue(-2 == 0 - 2)
@@ -30,8 +48,6 @@ class UnaryOpTestCase(unittest.TestCase):
         self.assertTrue(eval("~" + nines) == ~(10**32-1))
 
     def test_negation_of_exponentiation(self):
-        # Make sure '**' does the right thing; these form a
-        # regression test for SourceForge bug #456756.
         self.assertEqual(-2 ** 3, -8)
         self.assertEqual((-2) ** 3, -8)
         self.assertEqual(-2 ** 4, -16)
@@ -47,4 +63,4 @@ class UnaryOpTestCase(unittest.TestCase):
 
 
 if __name__ == "__main__":
-    unittest.main()
+    run_tests()
