diff --git a/test/cpython/v3_13/test_richcmp.py b/test/cpython/v3_13/test_richcmp.py
index 5f449cdc05c..a588d361785 100644
--- a/test/cpython/v3_13/test_richcmp.py
+++ b/test/cpython/v3_13/test_richcmp.py
@@ -1,3 +1,21 @@
+# ======= BEGIN Dynamo patch =======
+# Owner(s): ["module: dynamo"]
+
+# ruff: noqa
+# flake8: noqa
+
+# Test copied from
+# https://raw.githubusercontent.com/python/cpython/refs/tags/v3.13.5/Lib/test/test_richcmp.py
+
+import sys
+import torch
+import torch._dynamo.test_case
+import unittest
+from torch._dynamo.test_case import CPythonTestCase
+from torch.testing._internal.common_utils import run_tests
+
+# ======= END DYNAMO PATCH =======
+
 # Tests for rich comparisons
 
 import unittest
@@ -93,7 +111,7 @@ opmap = {
     "ge": (lambda a,b: a>=b, operator.ge, operator.__ge__)
 }
 
-class VectorTest(unittest.TestCase):
+class VectorTest(CPythonTestCase):
 
     def checkfail(self, error, opname, *args):
         for op in opmap[opname]:
@@ -135,7 +153,7 @@ class VectorTest(unittest.TestCase):
                     # calls __bool__, which should fail
                     self.assertRaises(TypeError, bool, op(a, b))
 
-class NumberTest(unittest.TestCase):
+class NumberTest(CPythonTestCase):
 
     def test_basic(self):
         # Check that comparisons involving Number objects
@@ -188,7 +206,7 @@ class NumberTest(unittest.TestCase):
         self.checkvalue("gt", 1, 0, True )
         self.checkvalue("ge", 1, 0, True )
 
-class MiscTest(unittest.TestCase):
+class MiscTest(CPythonTestCase):
 
     def test_misbehavin(self):
         class Misb:
@@ -279,7 +297,7 @@ class MiscTest(unittest.TestCase):
                     test[0]()
 
 
-class DictTest(unittest.TestCase):
+class DictTest(CPythonTestCase):
 
     def test_dicts(self):
         # Verify that __eq__ and __ne__ work for dicts even if the keys and
@@ -304,7 +322,7 @@ class DictTest(unittest.TestCase):
             for op in opmap[opname]:
                 self.assertRaises(TypeError, op, imag1a, imag2)
 
-class ListTest(unittest.TestCase):
+class ListTest(CPythonTestCase):
 
     def test_coverage(self):
         # exercise all comparisons for lists
@@ -353,4 +371,4 @@ class ListTest(unittest.TestCase):
 
 
 if __name__ == "__main__":
-    unittest.main()
+    run_tests()
