diff --git a/aten/src/ATen/test/CMakeLists.txt b/aten/src/ATen/test/CMakeLists.txt
index 8b69e5cab..730e00256 100644
--- a/aten/src/ATen/test/CMakeLists.txt
+++ b/aten/src/ATen/test/CMakeLists.txt
@@ -4,6 +4,11 @@ IF (MSVC)
   ENDIF()
 ENDIF(MSVC)
 
+if(WITH_ROCM)
+  SET(CMAKE_C_COMPILER ${HIP_HIPCC_EXECUTABLE})
+  SET(CMAKE_CXX_COMPILER ${HIP_HIPCC_EXECUTABLE})
+endif()
+
 ADD_EXECUTABLE(scalar_test scalar_test.cpp)
 target_link_libraries(scalar_test ATen)
 
@@ -43,13 +48,13 @@ target_link_libraries(verify_api_visibility ATen)
 add_executable(tbb_init_test tbb_init_test.cpp)
 target_link_libraries(tbb_init_test ATen)
 
-if(NOT NO_CUDA)
-  cuda_add_executable(integer_divider_test integer_divider_test.cu)
+if(WITH_ROCM)
+  hip_add_executable(integer_divider_test integer_divider_test.cu)
   target_link_libraries(integer_divider_test ATen)
 endif()
 
-if(NOT NO_CUDA)
-  cuda_add_executable(cuda_rng_test cuda_rng_test.cpp)
+if(WITH_ROCM)
+  hip_add_executable(cuda_rng_test cuda_rng_test.cpp)
   target_link_libraries(cuda_rng_test ATen)
 endif()
 
