diff --git a/aten/src/TH/generic/simd/convolve5x5_sse.cpp b/aten/src/TH/generic/simd/convolve5x5_sse.cpp
index 9de9a4a..feef8ad 100644
--- a/aten/src/TH/generic/simd/convolve5x5_sse.cpp
+++ b/aten/src/TH/generic/simd/convolve5x5_sse.cpp
@@ -2,7 +2,7 @@
 #include "common_simd.h"
 #include <stdint.h>

-
+#if !defined(__HIP_PLATFORM_HCC__)
 /* SSE variants */
 void convolve_5x5_1_sse(float* output, float* image, float* weight, int64_t count, int64_t outputStride, int64_t inputStride) {
   int64_t i = 0;
@@ -319,3 +319,4 @@ void convolve_5x5_sse(float* output, float* input, float* kernel, int64_t outRow
     r_ += (outStride * 1);
   }
 }
+#endif
