diff --git a/aten/src/THC/THCAllocator.c b/aten/src/THC/THCAllocator.c
index 554a379d7..4ff8b0c55 100644
--- a/aten/src/THC/THCAllocator.c
+++ b/aten/src/THC/THCAllocator.c
@@ -64,7 +64,7 @@ static void *THCUVAAllocator_alloc(void* ctx, ptrdiff_t size) {
   // See J.1.1 of the CUDA_C_Programming_Guide.pdf for UVA and coherence rules
   // on various compute capabilities.
   void* ptr;
-  THCudaCheck(cudaMallocManaged(&ptr, size, cudaMemAttachGlobal));
+  throw std::runtime_error("HIP does not support MallocManaged.");
   return ptr;
 }
 
