diff --git a/aten/src/TH/THAtomic.c b/aten/src/TH/THAtomic.c
index 16f0ddb48..111e28364 100644
--- a/aten/src/TH/THAtomic.c
+++ b/aten/src/TH/THAtomic.c
@@ -24,7 +24,7 @@ void THAtomicSet(int32_t volatile *a, int32_t newvalue)
 #if defined(USE_C11_ATOMICS)
   atomic_store(a, newvalue);
 #elif defined(USE_MSC_ATOMICS)
-  assert(sizeof(int) == sizeof(int32_t));
+  
   _InterlockedExchange((int32_t*)a, newvalue);
 #elif defined(USE_GCC_ATOMICS)
   __sync_lock_test_and_set(a, newvalue);
