diff --git a/aten/src/ATen/cuda/CUDAHalf.cuh b/aten/src/ATen/cuda/CUDAHalf.cuh
index b5d9520a0..dde2f33e5 100644
--- a/aten/src/ATen/cuda/CUDAHalf.cuh
+++ b/aten/src/ATen/cuda/CUDAHalf.cuh
@@ -11,7 +11,7 @@ namespace at {
 template <> AT_API half convert(Half aten_half);
 template <> AT_API Half convert(half cuda_half);
 template <> AT_API half convert(double value);
-#if CUDA_VERSION >= 9000
+#if CUDA_VERSION >= 9000 || defined(__HIP_PLATFORM_HCC__)
 template <> __half HalfFix(Half h);
 template <> Half HalfFix(__half h);
 #endif
