diff --git a/aten/src/THCUNN/BCECriterion.cu b/aten/src/THCUNN/BCECriterion.cu
index ccb40008c..1051a19af 100644
--- a/aten/src/THCUNN/BCECriterion.cu
+++ b/aten/src/THCUNN/BCECriterion.cu
@@ -9,6 +9,14 @@
 #include <thrust/transform.h>
 #include <thrust/transform_reduce.h>
 
+#if defined(__HIP_PLATFORM_HCC__)
+template <typename T>
+inline __host__ __device__ T eps();
+template <>
+inline __host__ __device__ float eps() { return 1e-12f; }
+template <>
+inline __host__ __device__ double eps() { return 1e-12; }
+#else
 template <typename T>
 inline __device__ T eps();
 
@@ -17,6 +25,7 @@ inline __device__ float eps() { return 1e-12f; }
 
 template <>
 inline __device__ double eps() { return 1e-12; }
+#endif
 
 template <typename Dtype, typename Acctype>
 struct bce_functor
