diff --git a/aten/src/ATen/native/cuda/SpectralOps.cu b/aten/src/ATen/native/cuda/SpectralOps.cu
index 61e159e18..4258d33e2 100644
--- a/aten/src/ATen/native/cuda/SpectralOps.cu
+++ b/aten/src/ATen/native/cuda/SpectralOps.cu
@@ -20,9 +20,10 @@
 
 #include <thrust/execution_policy.h>
 #include <thrust/unique.h>
-#include <cufft.h>
-#include <cufftXt.h>
+
+
 #include <cmath>
+#if !defined(__HIP_PLATFORM_HCC__)
 
 namespace at { namespace native {
 
@@ -411,3 +412,5 @@ Tensor _fft_cufft(const Tensor& self, int64_t signal_ndim,
 }
 
 }} // at::native
+
+#endif
\ No newline at end of file
