diff --git a/aten/src/ATen/native/cuda/Embedding.cu b/aten/src/ATen/native/cuda/Embedding.cu
index 67b3265b2..148ae4f6d 100644
--- a/aten/src/ATen/native/cuda/Embedding.cu
+++ b/aten/src/ATen/native/cuda/Embedding.cu
@@ -165,11 +165,11 @@ __global__ void renorm_kernel(
   for (int i = tid; i < dim; i += blockDim.x) {
     auto x = scalar_cast<accscalar_t>(weights[base_index + i]);
     if (norm_type == 1) {
-      v += std::abs(x);
+      v += fabs(x);
     } else if (norm_type == 2) {
       v += x * x;
     } else {
-      v += std::pow(x, norm_type);
+      v += powf(x, norm_type);
     }
   }
 
@@ -177,7 +177,7 @@ __global__ void renorm_kernel(
   v = reduceBlock<accscalar_t>(sdata, blockDim.x, v, Op(), 0);
 
   if (tid == 0) {
-    sdata[0] = std::pow(v, scalar_cast<accscalar_t>(1.0 / norm_type));
+    sdata[0] = powf(v, scalar_cast<accscalar_t>(1.0 / norm_type));
   }
   __syncthreads();
 
