diff --git a/aten/src/TH/generic/simd/convolve5x5_avx.cpp b/aten/src/TH/generic/simd/convolve5x5_avx.cpp
index 560474b..1cc7bac 100644
--- a/aten/src/TH/generic/simd/convolve5x5_avx.cpp
+++ b/aten/src/TH/generic/simd/convolve5x5_avx.cpp
@@ -4,7 +4,7 @@


 #define CLEAR_AVX() _mm256_zeroupper()
-
+#if !defined(__HIP_PLATFORM_HCC__)
 void convolve_5x5_1_avx(float* output, float* image, float* weight, int64_t count, int64_t outputStride, int64_t inputStride) {
   int64_t i = 0;
   int64_t alignedCount = count & 0xFFFFFFF8;
@@ -211,4 +211,5 @@ void convolve_5x5_avx(float* output, float* input, float* kernel, int64_t outRow
     CLEAR_AVX();
     convolve_5x5_sse(&output[procCols], &input[procCols], kernel, outRows, remCols, outStride, inCols);
   }
-}
\ No newline at end of file
+}
+#endif
