diff --git a/aten/src/ATen/cuda/CUDAHalf.cu b/aten/src/ATen/cuda/CUDAHalf.cu
index dbb18fc3f..5c4754f58 100644
--- a/aten/src/ATen/cuda/CUDAHalf.cu
+++ b/aten/src/ATen/cuda/CUDAHalf.cu
@@ -4,6 +4,7 @@
 #include <cuda.h>
 #include <cuda_runtime.h>
 #include <cuda_fp16.h>
+#if !defined(__HIP_PLATFORM_HCC__)
 
 namespace at {
 #if CUDA_VERSION < 9000
@@ -54,3 +55,5 @@ template <> Half HalfFix(__half h) {
 }
 #endif
 } // namespace at
+
+#endif
\ No newline at end of file
