diff --git a/aten/src/THC/THCTensorTypeUtils.cuh b/aten/src/THC/THCTensorTypeUtils.cuh
index 78bea9746..61058dc3d 100644
--- a/aten/src/THC/THCTensorTypeUtils.cuh
+++ b/aten/src/THC/THCTensorTypeUtils.cuh
@@ -168,7 +168,7 @@ struct ScalarNegate<half> {
 template <>
 struct ScalarInv<half> {
   static __host__ __device__ half to(const half v) {
-#ifdef __CUDA_ARCH__
+#if defined (__CUDA_ARCH_) || defined(__HIP_PLATFORM_HCC__)
     return __float2half(1.0f / __half2float(v));
 #else
     float fv = THC_half2float(v);
