diff --git a/scripts/get_python_cmake_flags.py b/scripts/get_python_cmake_flags.py
index 90badacaf..0220d1e5a 100644
--- a/scripts/get_python_cmake_flags.py
+++ b/scripts/get_python_cmake_flags.py
@@ -8,7 +8,7 @@
 # if your installation is not being properly detected by CMake.
 #
 #   mkdir -p build && cd build
-#   cmake $(python ../scripts/get_python_cmake_flags.py) ..
+#   cmake $(python ../scripts/get_python_libs.py) ..
 #   make
 #
 
