diff --git a/torch/lib/THD/base/data_channels/DataChannelMPI.cpp b/torch/lib/THD/base/data_channels/DataChannelMPI.cpp
index 56e300eb1..62fa6e7c7 100644
--- a/torch/lib/THD/base/data_channels/DataChannelMPI.cpp
+++ b/torch/lib/THD/base/data_channels/DataChannelMPI.cpp
@@ -10,7 +10,7 @@
 #include <unordered_map>
 #include <iostream>
 
-#ifdef WITH_CUDA
+#ifdef WITH_ROCM
 #include <cuda_runtime.h>
 #endif
 
@@ -140,7 +140,7 @@ rank_type DataChannelMPI::getNumProcesses() {
 struct AutoGPU {
   AutoGPU(int new_device) {
     if (new_device == -1) return;
-#ifdef WITH_CUDA
+#ifdef WITH_ROCM
     cudaGetDevice(&device_);
     cudaSetDevice(new_device);
 #endif
@@ -148,7 +148,7 @@ struct AutoGPU {
 
   ~AutoGPU() {
     if (device_ == -1) return;
-#ifdef WITH_CUDA
+#ifdef WITH_ROCM
     cudaSetDevice(device_);
 #endif
   }
