diff --git a/torch/csrc/utils/auto_stream.h b/torch/csrc/utils/auto_stream.h
index b18f7edcb..616313d20 100644
--- a/torch/csrc/utils/auto_stream.h
+++ b/torch/csrc/utils/auto_stream.h
@@ -2,13 +2,13 @@
 
 // RAII structs to set CUDA stream
 
-#ifdef WITH_CUDA
+#ifdef WITH_ROCM
 #include <THC/THC.h>
 extern THCState* state;
 #endif
 
 struct AutoStream {
-#ifdef WITH_CUDA
+#ifdef WITH_ROCM
   explicit AutoStream(THCStream* stream)
     : original_stream(THCState_getStream(state))
   {
