diff --git a/torch/csrc/allocators.h b/torch/csrc/allocators.h
index 25b2ac986..b6c431427 100644
--- a/torch/csrc/allocators.h
+++ b/torch/csrc/allocators.h
@@ -5,7 +5,7 @@
 #include <memory>
 
 #include <TH/TH.h>
-#ifdef WITH_CUDA
+#ifdef WITH_ROCM
 #include <THC/THC.h>
 #endif
 
@@ -41,7 +41,7 @@ public:
   void free(void* ptr);
 };
 
-#ifdef WITH_CUDA
+#ifdef WITH_ROCM
 class CudaStorageWeakRefAllocator {
 public:
   CudaStorageWeakRefAllocator(PyObject *wrapped_object, THCDeviceAllocator *alloc, void *ctx) {
@@ -63,6 +63,6 @@ public:
 
 extern THAllocator THObjectPtrAllocator;
 extern THAllocator THStorageWeakRefAllocator;
-#ifdef WITH_CUDA
+#ifdef WITH_ROCM
 extern THCDeviceAllocator THCStorageWeakRefAllocator;
 #endif
