diff --git a/test/dynamo/cpython/3_13/test_operator.py b/test/dynamo/cpython/3_13/test_operator.py
index d90f820052c..cdfa02be429 100644
--- a/test/dynamo/cpython/3_13/test_operator.py
+++ b/test/dynamo/cpython/3_13/test_operator.py
@@ -1,3 +1,23 @@
+# ======= BEGIN Dynamo patch =======
+# Owner(s): ["module: dynamo"]
+
+# ruff: noqa
+# flake8: noqa
+
+# Test copied from
+# https://raw.githubusercontent.com/python/cpython/refs/tags/v3.13.5/Lib/test/test_operator.py
+
+import sys
+import torch
+import torch._dynamo.test_case
+import unittest
+from torch._dynamo.test_case import CPythonTestCase
+from torch.testing._internal.common_utils import run_tests
+
+__TestCase = CPythonTestCase
+
+# ======= END DYNAMO PATCH =======
+
 import unittest
 import inspect
 import pickle
@@ -62,7 +82,8 @@ class OperatorTestCase:
 
     def test_lt(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.lt)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.lt)
         self.assertRaises(TypeError, operator.lt, 1j, 2j)
         self.assertFalse(operator.lt(1, 0))
         self.assertFalse(operator.lt(1, 0.0))
@@ -73,7 +94,8 @@ class OperatorTestCase:
 
     def test_le(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.le)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.le)
         self.assertRaises(TypeError, operator.le, 1j, 2j)
         self.assertFalse(operator.le(1, 0))
         self.assertFalse(operator.le(1, 0.0))
@@ -84,10 +106,11 @@ class OperatorTestCase:
 
     def test_eq(self):
         operator = self.module
-        class C(object):
-            def __eq__(self, other):
-                raise SyntaxError
-        self.assertRaises(TypeError, operator.eq)
+        with torch._dynamo.error_on_graph_break(False):
+            class C(object):
+                def __eq__(self, other):
+                    raise SyntaxError
+            self.assertRaises(TypeError, operator.eq)
         self.assertRaises(SyntaxError, operator.eq, C(), C())
         self.assertFalse(operator.eq(1, 0))
         self.assertFalse(operator.eq(1, 0.0))
@@ -98,10 +121,11 @@ class OperatorTestCase:
 
     def test_ne(self):
         operator = self.module
-        class C(object):
-            def __ne__(self, other):
-                raise SyntaxError
-        self.assertRaises(TypeError, operator.ne)
+        with torch._dynamo.error_on_graph_break(False):
+            class C(object):
+                def __ne__(self, other):
+                    raise SyntaxError
+            self.assertRaises(TypeError, operator.ne)
         self.assertRaises(SyntaxError, operator.ne, C(), C())
         self.assertTrue(operator.ne(1, 0))
         self.assertTrue(operator.ne(1, 0.0))
@@ -112,7 +136,8 @@ class OperatorTestCase:
 
     def test_ge(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.ge)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.ge)
         self.assertRaises(TypeError, operator.ge, 1j, 2j)
         self.assertTrue(operator.ge(1, 0))
         self.assertTrue(operator.ge(1, 0.0))
@@ -123,7 +148,8 @@ class OperatorTestCase:
 
     def test_gt(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.gt)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.gt)
         self.assertRaises(TypeError, operator.gt, 1j, 2j)
         self.assertTrue(operator.gt(1, 0))
         self.assertTrue(operator.gt(1, 0.0))
@@ -134,26 +160,30 @@ class OperatorTestCase:
 
     def test_abs(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.abs)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.abs)
         self.assertRaises(TypeError, operator.abs, None)
         self.assertEqual(operator.abs(-1), 1)
         self.assertEqual(operator.abs(1), 1)
 
     def test_add(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.add)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.add)
         self.assertRaises(TypeError, operator.add, None, None)
         self.assertEqual(operator.add(3, 4), 7)
 
     def test_bitwise_and(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.and_)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.and_)
         self.assertRaises(TypeError, operator.and_, None, None)
         self.assertEqual(operator.and_(0xf, 0xa), 0xa)
 
     def test_concat(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.concat)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.concat)
         self.assertRaises(TypeError, operator.concat, None, None)
         self.assertEqual(operator.concat('py', 'thon'), 'python')
         self.assertEqual(operator.concat([1, 2], [3, 4]), [1, 2, 3, 4])
@@ -163,7 +193,8 @@ class OperatorTestCase:
 
     def test_countOf(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.countOf)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.countOf)
         self.assertRaises(TypeError, operator.countOf, None, None)
         self.assertRaises(ZeroDivisionError, operator.countOf, BadIterable(), 1)
         self.assertEqual(operator.countOf([1, 2, 1, 3, 1, 4], 3), 1)
@@ -177,33 +208,38 @@ class OperatorTestCase:
     def test_delitem(self):
         operator = self.module
         a = [4, 3, 2, 1]
-        self.assertRaises(TypeError, operator.delitem, a)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.delitem, a)
         self.assertRaises(TypeError, operator.delitem, a, None)
         self.assertIsNone(operator.delitem(a, 1))
         self.assertEqual(a, [4, 2, 1])
 
     def test_floordiv(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.floordiv, 5)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.floordiv, 5)
         self.assertRaises(TypeError, operator.floordiv, None, None)
         self.assertEqual(operator.floordiv(5, 2), 2)
 
     def test_truediv(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.truediv, 5)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.truediv, 5)
         self.assertRaises(TypeError, operator.truediv, None, None)
         self.assertEqual(operator.truediv(5, 2), 2.5)
 
     def test_getitem(self):
         operator = self.module
         a = range(10)
-        self.assertRaises(TypeError, operator.getitem)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.getitem)
         self.assertRaises(TypeError, operator.getitem, a, None)
         self.assertEqual(operator.getitem(a, 2), 2)
 
     def test_indexOf(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.indexOf)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.indexOf)
         self.assertRaises(TypeError, operator.indexOf, None, None)
         self.assertRaises(ZeroDivisionError, operator.indexOf, BadIterable(), 1)
         self.assertEqual(operator.indexOf([4, 3, 2, 1], 3), 1)
@@ -217,13 +253,15 @@ class OperatorTestCase:
 
     def test_invert(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.invert)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.invert)
         self.assertRaises(TypeError, operator.invert, None)
         self.assertEqual(operator.inv(4), -5)
 
     def test_lshift(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.lshift)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.lshift)
         self.assertRaises(TypeError, operator.lshift, None, 42)
         self.assertEqual(operator.lshift(5, 1), 10)
         self.assertEqual(operator.lshift(5, 0), 5)
@@ -231,13 +269,15 @@ class OperatorTestCase:
 
     def test_mod(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.mod)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.mod)
         self.assertRaises(TypeError, operator.mod, None, 42)
         self.assertEqual(operator.mod(5, 2), 1)
 
     def test_mul(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.mul)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.mul)
         self.assertRaises(TypeError, operator.mul, None, None)
         self.assertEqual(operator.mul(5, 2), 10)
 
@@ -245,14 +285,16 @@ class OperatorTestCase:
         operator = self.module
         self.assertRaises(TypeError, operator.matmul)
         self.assertRaises(TypeError, operator.matmul, 42, 42)
-        class M:
-            def __matmul__(self, other):
-                return other - 1
+        with torch._dynamo.error_on_graph_break(False):
+            class M:
+                def __matmul__(self, other):
+                    return other - 1
         self.assertEqual(M() @ 42, 41)
 
     def test_neg(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.neg)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.neg)
         self.assertRaises(TypeError, operator.neg, None)
         self.assertEqual(operator.neg(5), -5)
         self.assertEqual(operator.neg(-5), 5)
@@ -261,13 +303,15 @@ class OperatorTestCase:
 
     def test_bitwise_or(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.or_)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.or_)
         self.assertRaises(TypeError, operator.or_, None, None)
         self.assertEqual(operator.or_(0xa, 0x5), 0xf)
 
     def test_pos(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.pos)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.pos)
         self.assertRaises(TypeError, operator.pos, None)
         self.assertEqual(operator.pos(5), 5)
         self.assertEqual(operator.pos(-5), -5)
@@ -276,15 +320,17 @@ class OperatorTestCase:
 
     def test_pow(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.pow)
         self.assertRaises(TypeError, operator.pow, None, None)
         self.assertEqual(operator.pow(3,5), 3**5)
-        self.assertRaises(TypeError, operator.pow, 1)
-        self.assertRaises(TypeError, operator.pow, 1, 2, 3)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.pow)
+            self.assertRaises(TypeError, operator.pow, 1)
+            self.assertRaises(TypeError, operator.pow, 1, 2, 3)
 
     def test_rshift(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.rshift)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.rshift)
         self.assertRaises(TypeError, operator.rshift, None, 42)
         self.assertEqual(operator.rshift(5, 1), 2)
         self.assertEqual(operator.rshift(5, 0), 5)
@@ -292,7 +338,8 @@ class OperatorTestCase:
 
     def test_contains(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.contains)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.contains)
         self.assertRaises(TypeError, operator.contains, None, None)
         self.assertRaises(ZeroDivisionError, operator.contains, BadIterable(), 1)
         self.assertTrue(operator.contains(range(4), 2))
@@ -301,7 +348,8 @@ class OperatorTestCase:
     def test_setitem(self):
         operator = self.module
         a = list(range(3))
-        self.assertRaises(TypeError, operator.setitem, a)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.setitem, a)
         self.assertRaises(TypeError, operator.setitem, a, None, None)
         self.assertIsNone(operator.setitem(a, 0, 2))
         self.assertEqual(a, [2, 1, 2])
@@ -309,16 +357,18 @@ class OperatorTestCase:
 
     def test_sub(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.sub)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.sub)
         self.assertRaises(TypeError, operator.sub, None, None)
         self.assertEqual(operator.sub(5, 2), 3)
 
     def test_truth(self):
         operator = self.module
-        class C(object):
-            def __bool__(self):
-                raise SyntaxError
-        self.assertRaises(TypeError, operator.truth)
+        with torch._dynamo.error_on_graph_break(False):
+            class C(object):
+                def __bool__(self):
+                    raise SyntaxError
+            self.assertRaises(TypeError, operator.truth)
         self.assertRaises(SyntaxError, operator.truth, C())
         self.assertTrue(operator.truth(5))
         self.assertTrue(operator.truth([0]))
@@ -327,7 +377,8 @@ class OperatorTestCase:
 
     def test_bitwise_xor(self):
         operator = self.module
-        self.assertRaises(TypeError, operator.xor)
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.xor)
         self.assertRaises(TypeError, operator.xor, None, None)
         self.assertEqual(operator.xor(0xb, 0xc), 0x7)
 
@@ -335,22 +386,25 @@ class OperatorTestCase:
         operator = self.module
         a = b = 'xyzpdq'
         c = a[:3] + b[3:]
-        self.assertRaises(TypeError, operator.is_)
         self.assertTrue(operator.is_(a, b))
         self.assertFalse(operator.is_(a,c))
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.is_)
 
     def test_is_not(self):
         operator = self.module
         a = b = 'xyzpdq'
         c = a[:3] + b[3:]
-        self.assertRaises(TypeError, operator.is_not)
         self.assertFalse(operator.is_not(a, b))
         self.assertTrue(operator.is_not(a,c))
+        with torch._dynamo.error_on_graph_break(False):
+            self.assertRaises(TypeError, operator.is_not)
 
     def test_attrgetter(self):
         operator = self.module
-        class A:
-            pass
+        with torch._dynamo.error_on_graph_break(False):
+            class A:
+                pass
         a = A()
         a.name = 'arthur'
         f = operator.attrgetter('name')
@@ -371,9 +425,10 @@ class OperatorTestCase:
         self.assertEqual(operator.attrgetter('x','z','y')(record), ('X', 'Z', 'Y'))
         self.assertRaises(TypeError, operator.attrgetter, ('x', (), 'y'))
 
-        class C(object):
-            def __getattr__(self, name):
-                raise SyntaxError
+        with torch._dynamo.error_on_graph_break(False):
+            class C(object):
+                def __getattr__(self, name):
+                    raise SyntaxError
         self.assertRaises(SyntaxError, operator.attrgetter('foo'), C())
 
         # recursive gets
@@ -411,9 +466,10 @@ class OperatorTestCase:
         f = operator.itemgetter(10)
         self.assertRaises(IndexError, f, a)
 
-        class C(object):
-            def __getitem__(self, name):
-                raise SyntaxError
+        with torch._dynamo.error_on_graph_break(False):
+            class C(object):
+                def __getitem__(self, name):
+                    raise SyntaxError
         self.assertRaises(SyntaxError, operator.itemgetter(42), C())
 
         f = operator.itemgetter('name')
@@ -444,9 +500,10 @@ class OperatorTestCase:
         self.assertEqual(operator.itemgetter(slice(2, 4))(t), ('c', 'd'))
 
         # interesting sequences
-        class T(tuple):
-            'Tuple subclass'
-            pass
+        with torch._dynamo.error_on_graph_break(False):
+            class T(tuple):
+                'Tuple subclass'
+                pass
         self.assertEqual(operator.itemgetter(0)(T('abc')), 'a')
         self.assertEqual(operator.itemgetter(0)(['a', 'b', 'c']), 'a')
         self.assertEqual(operator.itemgetter(0)(range(100, 200)), 100)
@@ -455,13 +512,14 @@ class OperatorTestCase:
         operator = self.module
         self.assertRaises(TypeError, operator.methodcaller)
         self.assertRaises(TypeError, operator.methodcaller, 12)
-        class A:
-            def foo(self, *args, **kwds):
-                return args[0] + args[1]
-            def bar(self, f=42):
-                return f
-            def baz(*args, **kwds):
-                return kwds['name'], kwds['self']
+        with torch._dynamo.error_on_graph_break(False):
+            class A:
+                def foo(self, *args, **kwds):
+                    return args[0] + args[1]
+                def bar(self, f=42):
+                    return f
+                def baz(*args, **kwds):
+                    return kwds['name'], kwds['self']
         a = A()
         f = operator.methodcaller('foo')
         self.assertRaises(IndexError, f, a)
@@ -480,21 +538,22 @@ class OperatorTestCase:
 
     def test_inplace(self):
         operator = self.module
-        class C(object):
-            def __iadd__     (self, other): return "iadd"
-            def __iand__     (self, other): return "iand"
-            def __ifloordiv__(self, other): return "ifloordiv"
-            def __ilshift__  (self, other): return "ilshift"
-            def __imod__     (self, other): return "imod"
-            def __imul__     (self, other): return "imul"
-            def __imatmul__  (self, other): return "imatmul"
-            def __ior__      (self, other): return "ior"
-            def __ipow__     (self, other): return "ipow"
-            def __irshift__  (self, other): return "irshift"
-            def __isub__     (self, other): return "isub"
-            def __itruediv__ (self, other): return "itruediv"
-            def __ixor__     (self, other): return "ixor"
-            def __getitem__(self, other): return 5  # so that C is a sequence
+        with torch._dynamo.error_on_graph_break(False):
+            class C(object):
+                def __iadd__     (self, other): return "iadd"
+                def __iand__     (self, other): return "iand"
+                def __ifloordiv__(self, other): return "ifloordiv"
+                def __ilshift__  (self, other): return "ilshift"
+                def __imod__     (self, other): return "imod"
+                def __imul__     (self, other): return "imul"
+                def __imatmul__  (self, other): return "imatmul"
+                def __ior__      (self, other): return "ior"
+                def __ipow__     (self, other): return "ipow"
+                def __irshift__  (self, other): return "irshift"
+                def __isub__     (self, other): return "isub"
+                def __itruediv__ (self, other): return "itruediv"
+                def __ixor__     (self, other): return "ixor"
+                def __getitem__(self, other): return 5  # so that C is a sequence
         c = C()
         self.assertEqual(operator.iadd     (c, 5), "iadd")
         self.assertEqual(operator.iand     (c, 5), "iand")
@@ -520,9 +579,10 @@ class OperatorTestCase:
 
     def test_index(self):
         operator = self.module
-        class X:
-            def __index__(self):
-                return 1
+        with torch._dynamo.error_on_graph_break(False):
+            class X:
+                def __index__(self):
+                    return 1
 
         self.assertEqual(operator.index(X()), 1)
         self.assertEqual(operator.index(0), 0)
@@ -539,9 +599,10 @@ class OperatorTestCase:
 
     def test_not_(self):
         operator = self.module
-        class C:
-            def __bool__(self):
-                raise SyntaxError
+        with torch._dynamo.error_on_graph_break(False):
+            class C:
+                def __bool__(self):
+                    raise SyntaxError
         self.assertRaises(TypeError, operator.not_)
         self.assertRaises(SyntaxError, operator.not_, C())
         self.assertFalse(operator.not_(5))
@@ -551,15 +612,16 @@ class OperatorTestCase:
 
     def test_length_hint(self):
         operator = self.module
-        class X(object):
-            def __init__(self, value):
-                self.value = value
+        with torch._dynamo.error_on_graph_break(False):
+            class X(object):
+                def __init__(self, value):
+                    self.value = value
 
-            def __length_hint__(self):
-                if type(self.value) is type:
-                    raise self.value
-                else:
-                    return self.value
+                def __length_hint__(self):
+                    if type(self.value) is type:
+                        raise self.value
+                    else:
+                        return self.value
 
         self.assertEqual(operator.length_hint([], 2), 0)
         self.assertEqual(operator.length_hint(iter([1, 2, 3])), 3)
@@ -574,7 +636,8 @@ class OperatorTestCase:
         with self.assertRaises(LookupError):
             operator.length_hint(X(LookupError))
 
-        class Y: pass
+        with torch._dynamo.error_on_graph_break(False):
+            class Y: pass
 
         msg = "'str' object cannot be interpreted as an integer"
         with self.assertRaisesRegex(TypeError, msg):
@@ -628,11 +691,11 @@ class OperatorTestCase:
         self.assertEqual(str(sig), '(obj, /)')
 
 
-class PyOperatorTestCase(OperatorTestCase, unittest.TestCase):
+class PyOperatorTestCase(OperatorTestCase, __TestCase):
     module = py_operator
 
 @unittest.skipUnless(c_operator, 'requires _operator')
-class COperatorTestCase(OperatorTestCase, unittest.TestCase):
+class COperatorTestCase(OperatorTestCase, __TestCase):
     module = c_operator
 
 
@@ -645,8 +708,9 @@ class OperatorPickleTestCase:
 
     def test_attrgetter(self):
         attrgetter = self.module.attrgetter
-        class A:
-            pass
+        with torch._dynamo.error_on_graph_break(False):
+            class A:
+                pass
         a = A()
         a.x = 'X'
         a.y = 'Y'
@@ -688,13 +752,14 @@ class OperatorPickleTestCase:
 
     def test_methodcaller(self):
         methodcaller = self.module.methodcaller
-        class A:
-            def foo(self, *args, **kwds):
-                return args[0] + args[1]
-            def bar(self, f=42):
-                return f
-            def baz(*args, **kwds):
-                return kwds['name'], kwds['self']
+        with torch._dynamo.error_on_graph_break(False):
+            class A:
+                def foo(self, *args, **kwds):
+                    return args[0] + args[1]
+                def bar(self, f=42):
+                    return f
+                def baz(*args, **kwds):
+                    return kwds['name'], kwds['self']
         a = A()
         for proto in range(pickle.HIGHEST_PROTOCOL + 1):
             with self.subTest(proto=proto):
@@ -717,25 +782,25 @@ class OperatorPickleTestCase:
                 # Can't test repr consistently with multiple keyword args
                 self.assertEqual(f2(a), f(a))
 
-class PyPyOperatorPickleTestCase(OperatorPickleTestCase, unittest.TestCase):
+class PyPyOperatorPickleTestCase(OperatorPickleTestCase, __TestCase):
     module = py_operator
     module2 = py_operator
 
 @unittest.skipUnless(c_operator, 'requires _operator')
-class PyCOperatorPickleTestCase(OperatorPickleTestCase, unittest.TestCase):
+class PyCOperatorPickleTestCase(OperatorPickleTestCase, __TestCase):
     module = py_operator
     module2 = c_operator
 
 @unittest.skipUnless(c_operator, 'requires _operator')
-class CPyOperatorPickleTestCase(OperatorPickleTestCase, unittest.TestCase):
+class CPyOperatorPickleTestCase(OperatorPickleTestCase, __TestCase):
     module = c_operator
     module2 = py_operator
 
 @unittest.skipUnless(c_operator, 'requires _operator')
-class CCOperatorPickleTestCase(OperatorPickleTestCase, unittest.TestCase):
+class CCOperatorPickleTestCase(OperatorPickleTestCase, __TestCase):
     module = c_operator
     module2 = c_operator
 
 
 if __name__ == "__main__":
-    unittest.main()
+    run_tests()
