diff --git a/test/dynamo/cpython/3_13/test_userstring.py b/test/dynamo/cpython/3_13/test_userstring.py
index 74df52f5412..1cc7c1debbc 100644
--- a/test/dynamo/cpython/3_13/test_userstring.py
+++ b/test/dynamo/cpython/3_13/test_userstring.py
@@ -1,14 +1,65 @@
+# ======= BEGIN Dynamo patch =======
+# Owner(s): ["module: dynamo"]
+
+# ruff: noqa
+# flake8: noqa
+
+import sys
+import torch
+import torch._dynamo.test_case
+import unittest
+from torch._dynamo.test_case import CPythonTestCase
+from torch.testing._internal.common_utils import run_tests
+
+__TestCase = CPythonTestCase
+
+
+# redirect import statements
+import sys
+import importlib.abc
+
+redirect_imports = (
+    "test.mapping_tests",
+    "test.typinganndata",
+    "test.test_grammar",
+    "test.test_math",
+    "test.test_iter",
+    "test.typinganndata.ann_module",
+)
+
+class RedirectImportFinder(importlib.abc.MetaPathFinder):
+    def find_spec(self, fullname, path, target=None):
+        # Check if the import is the problematic one
+        if fullname in redirect_imports:
+            try:
+                # Attempt to import the standalone module
+                name = fullname.removeprefix("test.")
+                r = importlib.import_module(name)
+                # Redirect the module in sys.modules
+                sys.modules[fullname] = r
+                # Return a module spec from the found module
+                return importlib.util.find_spec(name)
+            except ImportError:
+                return None
+        return None
+
+# Add the custom finder to sys.meta_path
+sys.meta_path.insert(0, RedirectImportFinder())
+
+
+# ======= END DYNAMO PATCH =======
+
 # UserString is a wrapper around the native builtin string type.
 # UserString instances should behave similar to builtin string objects.
 
 import unittest
-from test import string_tests
+import string_tests
 
 from collections import UserString
 
 class UserStringTest(
     string_tests.StringLikeTest,
-    unittest.TestCase
+    __TestCase
     ):
 
     type2test = UserString
@@ -68,4 +119,4 @@ class UserStringTest(
 
 
 if __name__ == "__main__":
-    unittest.main()
+    run_tests()
